app/components/task-card.js

```diff
@@ -31,21 +43,134 @@ export default Component.extend({
   'data-model-id': alias('task.id'),
   'data-model-type': 'task',
 
-  assignedUser: alias('task.assignedUser'),
-
   isLoading: alias('task.isLoading'),
 
-  taskType: alias('task.taskType'),
-  taskTypeClass: computed('taskType', function() {
-    return `task-card--${get(this, 'taskType')}`;
+  selectedOption: computed('taskUser', 'users', function() {
+    let user = get(this, 'taskUser');
+    let users = get(this, 'users');
+    if (isEmpty(user) || isEmpty(users)) {
+      return;
+    }
+    return users.find((item) => {
+      return get(user, 'id') === get(item, 'id');
+    });
   }),
+
   taskSkills: mapBy('task.taskSkills', 'skill'),
 
-  click() {
+  users: alias('members'),
+
+  sortedUserOptions: sort('userOptions', function(a, b) {
+    let userId = get(this, 'currentUser.user.id');
+    let taskUserId = get(this, 'taskUser.id');
+    let aId = get(a, 'id');
+    let bId = get(b, 'id');
+    let aIsUser = aId === userId;
+    let bIsUser = bId === userId;
+    let aIsTaskUser = aId === taskUserId;
+    let bIsTaskUser = bId === taskUserId;
+
+    if (aIsUser) {
+      return -1; // move the user up
+    } else if (aIsTaskUser) {
+      return bIsUser ? 1 : -1; // move task user up unless user is above
+    } else if (bIsUser || bIsTaskUser) {
+      return 1; // keep user and task user up top
+    } else {
+      // sort by username
+      return get(a, 'username') < get(b, 'username') ? -1 : 1;
+    }
+  }),
+
+  userOptions: computed('results', 'users', 'taskUser', function() {
+    let results = get(this, 'results');
+    let users = get(this, 'users');
+
+    if (isEmpty(results)) {
+      return users.mapBy('content'); // TODO: Replace this!
+    } else {
+      return results.toArray();
+    }
+  }),
+
+  init() {
+    this._super(...arguments);
+    set(this, 'keyboardActivated', true);
+    set(this, 'boundKeyHandler', bind(this, this.keyHandler));
+  },
+
+  click(e) {
+    return false; // TODO: Remove this
     let isLoading = get(this, 'isLoading');
     let task = get(this, 'task');
     if (!isLoading) {
       this.sendAction('clickedTask', task);
     }
+  },
+
+  keyHandler(e) {
+    let key = getCode(e);
+    switch (key) {
+      case 'KeyA':
+        // TODO: Make this work
+        // this.showUsers();
+        // e.preventDefault();
+        break;
+      default:
+        break;
+    }
+  },
+
+  mouseEnter() {
+    this._bindKeydown();
+  },
+
+  mouseLeave() {
+    this._unbindKeydown();
+  },
+
+  showUsers() {
+    get(this, 'selectRemoteController').actions.open();
+    this._unbindKeydown();
+  },
+
+  willDestroyElement() {
+    this._unbindKeydown();
+  },
+
+  actions: {
+    async searchUsers(query) {
+      let store = get(this, 'store');
+      store.query('user', { query }).then((users) => {
+        set(this, 'results', users);
+      });
```

Comment:
To elaborate on the previous comment, if the function is`async`, then we could rewrite this as 

```Javascript
let users= await store.query('user', { query });
set(this, 'results', users);
```

If we don't want to write it that way, then marking the function `async` would be misleading as well, so I would remove it. @